src/cursor/cur_file.c

```diff
@@ -513,6 +547,15 @@ __curfile_create(WT_SESSION_IMPL *session,
 	/* Underlying btree initialization. */
 	__wt_btcur_open(cbt);
 
+	/*
+	 * WT_CURSOR.modify supported on 'u' value formats, but the fast-path
+	 * through the btree code requires log file format changes, it's not
+	 * available in all versions.
+	 */
+	if (WT_STREQ(cursor->value_format, "u") &&
+	    S2C(session)->compat_major >= WT_LOG_V2)
+		cursor->modify = __curfile_modify;
```

Comment:
Is this the only line needed to support compatibility versions?  That's amazing.